### PR TITLE
doc: links: group links by URL post-1.6.0

### PR DESCRIPTION
**Pull request by greg-fer**
Updated the links.txt to follow grouping guidelines.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
Related to: https://github.com/nrfconnect/sdk-nrfxlib/pull/486

**Review by b-gent — APPROVED** [review]

File: doc/nrf/links.txt [hunk 1]
```diff
@@ -537,6 +537,21 @@
 .. _`Memfault: Collecting Device Metrics`: https://docs.memfault.com/docs/embedded/metrics-api
 .. _`Memfault: Error Tracking with Trace Events`: https://docs.memfault.com/docs/embedded/trace-events/
 
+.. ### Source: nrfcloud.com
+
+.. _`nRF Cloud`:
```
Comment (peknis) [inline doc/nrf/links.txt:542]: I guess these empty items are just placeholders for the product name change?

Comment (greg-fer) [reply doc/nrf/links.txt:542]: That's more of a question to @melwee01 , but the placement of tags is juts a method for having two different tags use the same link. In this case, `nRF Cloud` and `nRF Connect for Cloud` will both link to https://nrfcloud.com/, and so on for other pairs.

Comment (peknis) [reply doc/nrf/links.txt:542]: Oh, that was new to me. Thanks for the hot tip Greg!

**Review by melwee01 — APPROVED** [review]

**Pull request merged**